### PR TITLE
vc: Remove bind destination when unmounting

### PR DESCRIPTION
**Pull request by darfux**
`virtcontainers.ensureDestinationExists` will create the bind destination directory/file, which should be removed properly when unmounting.

Fixes: https://github.com/kata-containers/runtime/issues/1974
Signed-off-by: Li Yuxuan <liyuxuan04@baidu.com>

### CODE REVIEW

File: virtcontainers/container.go [hunk 1]
```diff
@@ -612,6 +612,24 @@ func (c *Container) unmountHostMounts() error {
 				return err
 			}
 
+			if m.Type == "bind" {
+				s, err := os.Stat(m.HostPath)
+				if err != nil {
+					c.Logger().WithFields(logrus.Fields{
+						"host-path": m.HostPath,
+						"error":     err,
```
Comment (jodh-intel) [inline virtcontainers/container.go]: The error should be handled like this:

```go
c.Logger().WithError(err).WithFields(...)
```

But is this log call even required if the caller logs the error (I haven't checked)?

File: virtcontainers/container.go [hunk 2]
```diff
@@ -612,6 +612,24 @@ func (c *Container) unmountHostMounts() error {
 				return err
 			}
 
+			if m.Type == "bind" {
```
Comment (jodh-intel) [inline virtcontainers/container.go]: It would be great if you could add a unit test for this scenario to ` virtcontainers/container_test.go`.

Comment (darfux) [reply virtcontainers/container.go]: Thanks for the suggestion. I'll work on it 😀

File: virtcontainers/container.go (hunk 1 above)
Comment (darfux) [reply virtcontainers/container.go]: > The error should be handled like this

I copied this log line from the prev codes in the func:
```golang
if err := syscall.Unmount(m.HostPath, syscall.MNT_DETACH); err != nil {
  c.Logger().WithFields(logrus.Fields{
    "host-path": m.HostPath,
    "error":     err,
  }).Warn("Could not umount")
  return err
```
So is it an old style and should be `WithError(err)` now?

> is this log call even required

I also wonder why does `syscall.Unmount` do such log here... maybe it includes the `m.HostPath` which isn't carried by the error?

**Comment by darfux** [comment]
@jodh-intel I have used `errors.Wrapf` instead of the logger and added a unit test, PTAL :)

**Comment by darfux** [comment]
Hi @jodh-intel, how can I restart the CI test? I have tried re-pushing and commenting with /test but it seems that both don't work..

**Comment by grahamwhaley** [comment]
thanks @nitkon . @darfux - you might not have enought 'rights' on the github org to re-trigger. There are maybe three options to re-trigger:
- issue a `/ t e s t` or `/ r e t e s t`
- issue a `/ t e s t - u b u n t u` or similar if you need to trigger just one CI, and it has an individual trigger phrase
- if re-building a single failed test, follow the *details* link to the Jenkins page for that job, log into Jenkins (it uses your github acct details), and click 'rebuild' for that job. This also requires you have enough 'rights' in the kata github organisation etc.

**Comment by darfux** [comment]
@grahamwhaley Thanks for the explanation 😀

**Pull request merged**